docs/source/basics/building_a_pipeline.md

```diff
@@ -0,0 +1,257 @@
+  └─ morpheus.MultiMessage -> morpheus.MessageMeta
+Added stage: <to-file-3; WriteToFileStage(filename=.tmp/temp_out.json, overwrite=True, file_type=FileTypes.Auto, include_index_col=True)>
+  └─ morpheus.MessageMeta -> morpheus.MessageMeta
+====Pipeline Complete====
+Pipeline visualization saved to .tmp/simple_identity.png
+```
+
+### Pipeline Build Checks
+After the `====Building Pipeline====` message, if logging is `INFO` or greater, the CLI will print a list of all stages and the type transformations of each stage. To be a valid Pipeline, the output type of one stage must match the input type of the next. Many stages are flexible and will determine their type at runtime but some stages require a specific input type. If your Pipeline is configured incorrectly, Morpheus will report the error. For example, if we run the same command as above but forget the `serialize` stage:
+```bash
+morpheus --log_level=DEBUG run pipeline-nlp \
+  from-file --filename=examples/data/pcap_dump.jsonlines \
+  deserialize \
+  to-file --overwrite --filename .tmp/temp_out.json
+```
+
+Then the following error will be displayed:
+```
+RuntimeError: The to-file stage cannot handle input of <class 'morpheus.messages.multi_message.MultiMessage'>. Accepted input types: (<class 'morpheus.messages.message_meta.MessageMeta'>,)
+```
+
+This indicates that the ``to-file`` stage cannot accept the input type of `morpheus.pipeline.messages.MultiMessage`.
+This is because the ``to-file`` stage has no idea how to write that class to a file, it only knows how to write instances of `morpheus.messages.message_meta.MessageMeta`. To ensure you have a valid pipeline, examine the `Accepted input types: (<class 'morpheus.messages.message_meta.MessageMeta'>,)` portion of the message. This indicates you need a stage that converts from the output type of the `deserialize` stage, `MultiMessage`, to `MessageMeta`, which is exactly what the `serialize` stage does.
+
+### Kafka Source Example
+The above example essentially just copies a file. However it is an important to note that most Morpheus pipelines are similar in structure, in that they begin with a source stage (`from-file`) followed by a `deserialize` stage, end with a `serialize` stage followed by a sink stage (`to-file`), with the actual training or inference logic occurring in between.
+
+We could also easily swap out the source or sink stages in the above example without any impact to the pipeline as a whole. For example, to read from a Kafka topic, simply replace the `from-file` stage with `from-kafka`:
+
+> **Note**: This assumes a Kafka broker running on the localhost listening to port 9092. For testing Morpheus with Kafka refer to the [Quick Launch Kafka Cluster](../../docs/source/developer_guide/contributing.md#quick-launch-kafka-cluster) section of [contributing.md](../../docs/source/developer_guide/contributing.md) following steps 1-6 and then replace port `9092` with the port your Kafka instance is listening on.
```

Comment:
It appears that links to `Quick Launch Kafka Cluster` and `contributing.md` were broken when the documentation was reorganized.

Reply:
Another good catch!